### DISCUSSION TITLE
Repeating a Sequence

### DISCUSSION THREAD

**Discussion by KimSilvaA** [bonsai-rx/bonsai#1380] [Q&A] [answered]
Hi again,

I would like to use Bonsai to calculate latency for an experiment. I press a key to start a trial and would like to know when the mouse crosses an ROI and when it interacts with a pot (which I take with a key press). The issue is that I would like to repeat this sequence for each trial but I'm having trouble doing so. I tried inserting the Repeat Node in the end with no luck. I also tried to accumulate the times it's in the ROI and take the timestamp for the first instance. 

 I attached a picture of the setup for clarity. Thanks in advanced! 
[Latency.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/11536358/Latency.bonsai.zip)

![ROI Testing Area](https://github.com/bonsai-rx/bonsai/assets/124469197/4434d3cc-fd6e-4556-8523-64763acf08a2)

**Comment by bruno-f-cruz** [comment]
You have to end a sequence in order to repeat it:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/2681a1b5-6346-4f67-880b-96b16b2e6862)

**Reply by KimSilvaA** [reply]
Hi Bruno,

Thanks. I tried inserting the repeat node at the end of the sequence but the timestamp keeps updating regardless if I pressed the key to start the trial. Do you recommend another way of getting the latency? I tried using select many, window trigger and branching the sequence further but nothing works so far.

**Reply by bruno-f-cruz** [reply, marked as answer]
Declare your subjects connected to the KeyUp as PublishSubjects instead of behaviorSubjects. That should fix it. Since repeat resubscribes to the subject, if the type of the subject has memory it will send the last available value thereby triggering the SkipUntil operator.

**Reply by KimSilvaA** [reply]
It works now! Thank you so much Bruno 💯
